src/app/shared/auth.guard.ts

```diff
@@ -17,15 +17,35 @@
 import { Injectable } from '@angular/core';
 import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot } from '@angular/router';
 import { AuthService } from 'ng2-ui-auth';
+import { of } from 'rxjs/internal/observable/of';
+import { catchError, map } from 'rxjs/operators';
+import { LogoutService } from './logout.service';
+import { UsersService } from './openapi';
 
 @Injectable()
 export class AuthGuard implements CanActivate {
-  constructor(private auth: AuthService, private router: Router) {}
+  constructor(
+    private auth: AuthService,
+    private router: Router,
+    private logoutService: LogoutService,
+    private usersService: UsersService
+  ) {}
   canActivate(next: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
-    if (this.auth.isAuthenticated()) {
-      return true;
-    }
-    this.router.navigate(['/login']);
-    return false;
+    return this.usersService.getUser().pipe(
```

Comment:
Just noting that this is going to block the auth pages load until the API request completes, which is ~90-100ms when I tried it just now against prod. Which isn't a lot, but it adds up.

OTOH, it's the right thing to do, and we could come up with some optimization later.